python/mpt/src/mpt/utils.cairo

```diff
@@ -189,8 +189,41 @@ func check_branch_node(node: BranchNode) {
             raise('ValueError');
         }
     }
+
     return ();
 }
+
+func check_leaf_node(path: Bytes, node: LeafNode) {
+    alloc_locals;
+
+    let nibbles_len = node.value.rest_of_key.value.len;
+    let path_len = path.value.len;
+
+    if (nibbles_len + path_len != 64) {
+        raise('ValueError');
+    }
+
+    let bytes_variant = node.value.value.value.bytes.value;
+    if (cast(bytes_variant, felt) != 0) {
+        if (bytes_variant.len != 0) {
```

Comment:
what are you asserting here? that the leaf node value is not null?
looks like a typo

Reply:
yes, asserting that the leaf value has a value?

Reply:
if the bytes pointer is not null and the bytes len is not zero, then raise?

Reply:
maybe 
```
if (cast(bytes_variant, felt) != 0) {
        if (bytes_variant.len == 0) {
            raise('ValueError');
        }
}

Reply:
🤦 omg